### DISCUSSION TITLE
How to use Jacoco quarkus dependency for code coverage in multi module project?

### DISCUSSION THREAD

**Discussion by JakubKubena** [quarkusio/quarkus#30497] [Q&A]
I have a fresh 2 module quarkus project with quarkus jacoco dependency. Unit tests for both modules are in the second one and whatever I do, I cant get coverage on the module which does not include tests. From reading the quarkus jacoco guide, my understanding was that this should work out of the box, without any extensive configuration. What am I missing?

**Comment by stephan-strate** [comment]
Do your unit tests use `@QuarkusTest`? If not, you will need to follow the steps here: https://quarkus.io/guides/tests-with-coverage#coverage-for-tests-not-using-quarkustest

Quarkus Jacoco does work out of the box for me, if I follow the steps in the guide precisely. Every step is important.

**Reply by JakubKubena** [reply]
Yes. All my tests use the `@QuarkusTest` annotation. Does your project have multiple modules, where you have the unit tests for all modules in just a single module?

There isnt much to follow in the guide since the only relevant step is adding the quarkus jacoco dependency, because I dont have integration tests.

**Comment by derari** [comment]
I am facing the same issue.

If I *don't* annotate with @QuarkusTest, the coverage for the included module is computed correctly, but *with* the @QuarkusTest annotation, no coverage for the other module is computed

**Reply by derari** [reply]
Adding a beans.xml or creating a jandex index for the other module fixed the issue for me.

I think this should be added to the documentation, if nothing else is done about it.

**Reply by AndreasPetersen** [reply]
> Adding a beans.xml or creating a jandex index for the other module fixed the issue for me.
> 
> I think this should be added to the documentation, if nothing else is done about it.

Thanks @derari , you really saved the day! I agree with you that this should be added to the https://quarkus.io/guides/tests-with-coverage documentation.
